Examples of software at each tier (#427)
EVERSE-ResearchSoftware/RSQKit: pages/research_software/three_tier_view.md

The **three-tier model of research software** provides a framework for understanding the diverse landscape of software 

> in research - from scripts, code, notebooks, to computational workflows, libraries, modules, frameworks, utilities and applications. 

Reviewer: ```suggestion
in research - from scripts, code, notebooks, to computational workflows, libraries, modules, frameworks, utilities and applications. This model distinguishes three tiers of research software, based on the intent of the research software - i.e. focussing on the purpose for which the software is being built:
```

Author: Oh will add this comment separately - looks like I missed it the first time.

> The three-tier model acknowledges a relationship between the tiers: software at higher tiers builds on the existence and stability of software in lower tiers,

Reviewer: That's not true - the analysis script might be build on a library which is considered infrastructure - I would get rid of this line and the following two lines also - unless they came from any reference docs.

> while the existence of higher-tier software reinforces the value of lower-tier software.

Reviewer: ```
```

*Diagram from ["EVERSE Paving the way towards a European Virtual Institute for Research Software Excellence" presentation](https://indico.cern.ch/event/1501988/contributions/6323204/attachments/3016679/5320601/EVERSE_Overview_Slides.pdf) by F. Psomopoulos, February 2025, adapted by Aleksandra Nenadic under CC-BY 4.0 licence*

## Analysis code

Analysis code is research software that captures computational research processes and methodology, often used in simulation, data generation, preparation, analysis and visualisation. 
It typically represents software created for personal use with a small scope, such as scripts quickly put together to analyse data.

Small analysis scripts, especially in academic research or industry projects, often remain unpublished and are only found on the personal computers of their authors.

> As a consequence, results can't be easily verified or extended, scripts might be lost when a computer is wiped or a researcher leaves an institution, others may redo work that already exists but isn’t discoverable, and community growth and collaboration are hindered.

Reviewer: ```
```

> As a consequence, results can't be easily verified or extended, scripts might be lost when a computer is wiped or a researcher leaves an institution, others may redo work that already exists but isn’t discoverable, and community growth and collaboration are hindered.

Reviewer: ```suggestion
```

However, the trend toward publishing code alongside academic papers has been growing. 
More researchers are realising that sharing their analysis code alongside their results is essential for transparency, reproducibility, and collaboration.

> When shared and collaboratively developed, analysis scripts can evolve from small, specific solutions into more comprehensive prototype or full-fledged software infrastructure tools that benefit many people in the research or software development community. 

Reviewer: ```suggestion
When shared and collaboratively developed, analysis scripts developed for specific solutions can start on the journey towards becoming more comprehensive prototype or full-fledged software infrastructure tools that benefit many people in the research or software development community. 
```

> For some examples of analysis code, see:

Reviewer: ```suggestion
Here are some examples of analysis code and their associated publications:
```
